[connectivity_plus] Upgrading to version 2.0.1 causes 'error: lambda expressions are not supported in -source 7'

## System info

Issue occurs on: Android
Plugin name: connectivity_plus
Plugin version: 2.0.1

Flutter doctor output:

```
[✓] Flutter (Channel stable, 2.5.2, on macOS 11.6 darwin-arm)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 2020.3)
[✓] VS Code (version 1.60.2)
[✓] Connected device (2 available)
```

## Steps to Reproduce
Installing version 2.0.1 gives me the following error when running 'flutter build appbundle --release':

```
.pub-cache/hosted/pub.dartlang.org/connectivity_plus-2.0.1/android/src/main/java/dev/fluttercommunity/plus/connectivity/ConnectivityBroadcastReceiver.java:86: error: lambda expressions are not supported in -source 7
    Runnable runnable = () -> events.success(connectivity.getNetworkType());
                           ^
  (use -source 8 or higher to enable lambda expressions)
1 error

FAILURE: Build failed with an exception.
```

## Steps taken
The following steps have been taken before running the build command:
1) flutter clean
2) rm -rf .pub-cache/
3) flutter pub get
4) flutter build appbundle --release

Changing to version 1.0.4 of the plugin and re-running the above steps solves the issue.

--- a/packages/connectivity_plus/connectivity_plus/CHANGELOG.md
+++ b/packages/connectivity_plus/connectivity_plus/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 2.0.2
+
+- Fix for error: lambda expressions are not supported in -source 7
+
 ## 2.0.1
 
 - Upgrade Android compile SDK version

--- a/packages/connectivity_plus/connectivity_plus/android/build.gradle
+++ b/packages/connectivity_plus/connectivity_plus/android/build.gradle
@@ -24,6 +24,11 @@ apply plugin: 'com.android.library'
 android {
     compileSdkVersion 31
 
+    compileOptions {
+      sourceCompatibility JavaVersion.VERSION_1_8
+      targetCompatibility JavaVersion.VERSION_1_8
+    }
+
     defaultConfig {
         minSdkVersion 16
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"

--- a/packages/connectivity_plus/connectivity_plus/pubspec.yaml
+++ b/packages/connectivity_plus/connectivity_plus/pubspec.yaml
@@ -1,6 +1,6 @@
 name: connectivity_plus
 description: Flutter plugin for discovering the state of the network (WiFi & mobile/cellular) connectivity on Android and iOS.
-version: 2.0.1
+version: 2.0.2
 homepage: https://plus.fluttercommunity.dev/
 repository: https://github.com/fluttercommunity/plus_plugins/tree/main/packages/
 

